oauth2client/client.py

```diff
@@ -1222,6 +1229,31 @@ def create_scoped(self, scopes):
         """
         return self
 
+    @classmethod
+    def from_json(cls, s):
+        # This import needs to be here to avoid a circular dependency.
```

Comment:
I agree with you in the large because most TODOs don't refer the reader to a more authoritative place where the work item is tracked. The virtue of `TODO(<issue citation>):` is that it tells the reader of the code (who may not be familiar with every issue currently open in the issue tracker) that the problem they are seeing in the code is already considered a problem by the code's maintainers and tells them exactly where to find the specific issue covering the problem they're seeing.

@dlorenc: the problem with "This import needs to be here to avoid a circular dependency" is that it's tone-positive. It sounds like the speaker is saying "circular dependencies are usually wrong but here this is the right thing". A TODO here, in addition to the benefit laid out above, is that it's tone-negative - it tells the reader "we know this is a problem, just like circular dependencies are a problem in any code".